python-black.el

```diff
@@ -119,6 +119,22 @@ DISPLAY-ERRORS is non-nil, shows a buffer if the formatting fails."
   (and (= (point-min) beg)
        (= (point-max) end)))
 
+(defun python-black-block ()
+  "Reformats the current org-mode source block."
+  (interactive)
+  (save-mark-and-excursion
```

Comment:
this fiddles with text matching and `org-mode` internals. is there really no better way? what about something like this? https://emacs.stackexchange.com/a/48159

Reply:
The matching part of the function comes from the package ober-eval-in-repl. I'm not sure there's a better way. The function you point to is useful for results of source blocks, which is not the contents of a source block.

Reply:
It does work quite well using the beg and end matching that are used inside org-src--contents-area (without a need to make an excursion). I don't have much experience with org programming, but I don't see other easy functions to quickly match beginning and end of a source block than the (org-element-property) function.